### PR TITLE
rgw: move RGWObjManifest stuff into rgw_manifest.cc,h

### PR DESCRIPTION
**Pull request by cbodley**
moves more code out of rgw_rados.*

### CODE REVIEW

**Comment by cbodley** [comment]
shaman builds failing because i didn't update src/test/rgw/test_rgw_common.h. tests will need to construct a zone service

File: src/rgw/rgw_manifest.h [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+    is_raw = rhs.is_raw;
+    if (is_raw) {
+      raw_obj = rhs.raw_obj;
+    } else {
+      obj = rhs.obj;
+    }
+  }
+
+  rgw_raw_obj get_raw_obj(const RGWZoneGroup& zonegroup, const RGWZoneParams& zone) const;
+  rgw_raw_obj get_raw_obj(RGWSI_Zone* zone_svc) const;
```
Comment (cbodley) [inline src/rgw/rgw_manifest.h]: i wanted to make this `const RGWSI_Zone*`, but doing that got messy and i split those changes into https://github.com/ceph/ceph/pull/25412

**Comment by cbodley** [comment]
had trouble getting this tested because of conflicts with https://github.com/ceph/ceph/pull/25412; now that it merged, i rebased and will retest

**Comment by ivancich** [comment]
jenkins test make check arm64

**Comment by cbodley** [comment]
closing for now

**Pull request closed without merging**